File: invesdwin-context-parent/invesdwin-context-jruby/src/test/java/de/invesdwin/context/jruby/tests/callback/ParametersAndReturnsTestDoubleNan.rb
puts("getDouble")
unless (defined?(getDouble)).nil? || getDouble.nil?
	raise "getDouble already defined!"
end
getDouble = callback("getDouble")
getDoubleType = getDouble.class
puts(getDoubleType)
puts(getDouble)
if(getDoubleType != Float)
	raise "getDouble not Float!"
end
if(!getDouble.nan?)
	raise "getDouble not NaN!"
end
callback("setDouble", getDouble)

puts("getDoubleVector")
unless (defined?(getDoubleVector)).nil? || getDoubleVector.nil?
	raise "getDoubleVector already defined!"
end
getDoubleVector = callback("getDoubleVector")
getDoubleVectorType = getDoubleVector[0].class
puts(getDoubleVectorType)
puts(getDoubleVector)
if(getDoubleVectorType != Float)
	raise "getDoubleVector not Float!"
end
if(!getDoubleVector[1].nan?)
	raise "getDoubleVector[1] not NaN!"
end
callback("setDoubleVector", getDoubleVector)

puts("getDoubleVectorAsList")
unless (defined?(getDoubleVectorAsList)).nil? || getDoubleVectorAsList.nil?
	raise "getDoubleVectorAsList already defined!"
end
getDoubleVectorAsList = callback("getDoubleVectorAsList")
getDoubleVectorAsListType = getDoubleVectorAsList[0].class
puts(getDoubleVectorAsListType)
puts(getDoubleVectorAsList)
if(getDoubleVectorAsListType != Float)
	raise "getDoubleVectorAsList not Float!"
end
if(!getDoubleVectorAsList[1].nan?)
	raise "getDoubleVectorAsList[1] not NaN!"
end
callback("setDoubleVectorAsList", getDoubleVectorAsList)

puts("getDoubleMatrix")
unless (defined?(getDoubleMatrix)).nil? || getDoubleMatrix.nil?
	raise "getDoubleMatrix already defined!"
end
getDoubleMatrix = callback("getDoubleMatrix")
getDoubleMatrixType = getDoubleMatrix[0][0].class
puts(getDoubleMatrixType)
puts(getDoubleMatrix)
if(getDoubleMatrixType != Float)
	raise "getDoubleMatrix not Float!"
end
if(!getDoubleMatrix[0][0].nan?)
	raise "getDoubleMatrix[0][0] not NaN!"
end
if(!getDoubleMatrix[1][1].nan?)
	raise "getDoubleMatrix[1][1] not NaN!"
end
if(!getDoubleMatrix[2][2].nan?)
	raise "getDoubleMatrix[2][2] not NaN!"
end
callback("setDoubleMatrix", getDoubleMatrix)

puts("getDoubleMatrixAsList")
unless (defined?(getDoubleMatrixAsList)).nil? || getDoubleMatrixAsList.nil?
	raise "getDoubleMatrixAsList already defined!"
end
getDoubleMatrixAsList = callback("getDoubleMatrixAsList")
getDoubleMatrixAsListType = getDoubleMatrixAsList[0][0].class
puts(getDoubleMatrixAsListType)
puts(getDoubleMatrixAsList)
if(getDoubleMatrixAsListType != Float)
	raise "getDoubleMatrixAsList not Float!"
end
if(!getDoubleMatrixAsList[0][0].nan?)
	raise "getDoubleMatrixAsList[0][0] not NaN!"
end
if(!getDoubleMatrixAsList[1][1].nan?)
	raise "getDoubleMatrixAsList[1][1] not NaN!"
end
if(!getDoubleMatrixAsList[2][2].nan?)
	raise "getDoubleMatrixAsList[2][2] not NaN!"
end
callback("setDoubleMatrixAsList", getDoubleMatrixAsList)

File: invesdwin-context-parent/invesdwin-context-javascript/src/test/java/de/invesdwin/context/javascript/tests/callback/ParametersAndReturnsTestDoubleNan.js
print("getDouble")
if(typeof getDouble !== 'undefined')
	throw "getDouble already defined!"
getDouble = callback("getDouble")
getDoubleType = typeof getDouble
print(getDoubleType)
print(getDouble)
if(getDoubleType !== 'number')
	throw "getDouble not number!"
if(!isNaN(getDouble))
	throw "getDouble not NaN!"
callback("setDouble",getDouble)

print("getDoubleVector")
if(typeof getDoubleVector !== 'undefined')
	throw "getDoubleVector already defined!"
getDoubleVector = callback("getDoubleVector")
getDoubleVectorType = typeof getDoubleVector[0]
print(getDoubleVectorType)
print(getDoubleVector)
if(getDoubleVectorType !== 'number')
	throw "getDoubleVector not number!"
if(!isNaN(getDoubleVector[1]))
	throw "getDoubleVector[1] not NaN!"
callback("setDoubleVector",getDoubleVector)

print("getDoubleVectorAsList")
if(typeof getDoubleVectorAsList !== 'undefined')
	throw "getDoubleVectorAsList already defined!"
getDoubleVectorAsList = callback("getDoubleVectorAsList")
getDoubleVectorAsListType = typeof getDoubleVectorAsList[0]
print(getDoubleVectorAsListType)
print(getDoubleVectorAsList)
if(getDoubleVectorAsListType !== 'number')
	throw "getDoubleVectorAsList not number!"
if(!isNaN(getDoubleVectorAsList[1]))
	throw "getDoubleVectorAsList[1] not NaN!"
callback("setDoubleVectorAsList",getDoubleVectorAsList)

print("getDoubleMatrix")
if(typeof getDoubleMatrix !== 'undefined')
	throw "getDoubleMatrix already defined!"
getDoubleMatrix = callback("getDoubleMatrix")
getDoubleMatrixType = typeof getDoubleMatrix[0][0]
print(getDoubleMatrixType)
print(getDoubleMatrix)
if(getDoubleMatrixType !== 'number')
	throw "getDoubleMatrix not number!"
if(!isNaN(getDoubleMatrix[0][0]))
	throw "getDoubleMatrix[0][0] not NaN!"
if(!isNaN(getDoubleMatrix[1][1]))
	throw "getDoubleMatrix[1][1] not NaN!"
if(!isNaN(getDoubleMatrix[2][2]))
	throw "getDoubleMatrix[2][2] not NaN!"
callback("setDoubleMatrix",getDoubleMatrix)

print("getDoubleMatrixAsList")
if(typeof getDoubleMatrixAsList !== 'undefined')
	throw "getDoubleMatrixAsList already defined!"
getDoubleMatrixAsList = callback("getDoubleMatrixAsList")
getDoubleMatrixAsListType = typeof getDoubleMatrixAsList[0][0]
print(getDoubleMatrixAsListType)
print(getDoubleMatrixAsList)
if(getDoubleMatrixAsListType !== 'number')
	throw "getDoubleMatrixAsList not number!"
if(!isNaN(getDoubleMatrixAsList[0][0]))
	throw "getDoubleMatrixAsList[0][0] not NaN!"
if(!isNaN(getDoubleMatrixAsList[1][1]))
	throw "getDoubleMatrixAsList[1][1] not NaN!"
if(!isNaN(getDoubleMatrixAsList[2][2]))
	throw "getDoubleMatrixAsList[2][2] not NaN!"
callback("setDoubleMatrixAsList",getDoubleMatrixAsList)
